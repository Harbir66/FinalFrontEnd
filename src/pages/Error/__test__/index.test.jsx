import React from 'react';
import { render } from '@testing-library/react';
import Error from '../index';

describe('Error Page', () => {
  it('should render correctly withour status code', () => {
    const { asFragment } = render(<Error />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correctly with status code', () => {
    jest.mock('react-router-dom', () => ({
      useParams: () => ({
        errorCode: 404,
      }),
    }));
    const { asFragment } = render(<Error />);
    expect(asFragment()).toMatchSnapshot();
  });
});
