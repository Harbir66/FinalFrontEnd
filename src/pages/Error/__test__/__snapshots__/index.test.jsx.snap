// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page should render correctly with status code 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="sidebar"
    >
      <div
        class="sidebar-header"
      >
        CMS+
      </div>
      <div
        class="sidebar-search"
      >
        <span>
          COLLECTION TYPES
        </span>
        <img
          alt="searchIcon"
          src="icon-search-dark@3x.png"
        />
      </div>
      <nav>
        <ul />
      </nav>
      <button
        style="background: black;"
        type="button"
      >
        CONTENT TYPE BUILDER
      </button>
    </div>
    <div
      class="errorContainer"
    >
      <header
        class="header"
      >
        Error
      </header>
      <div
        class="content"
      >
        <p>
          Something went wrong!
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Error Page should render correctly withour status code 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="sidebar"
    >
      <div
        class="sidebar-header"
      >
        CMS+
      </div>
      <div
        class="sidebar-search"
      >
        <span>
          COLLECTION TYPES
        </span>
        <img
          alt="searchIcon"
          src="icon-search-dark@3x.png"
        />
      </div>
      <nav>
        <ul />
      </nav>
      <button
        style="background: black;"
        type="button"
      >
        CONTENT TYPE BUILDER
      </button>
    </div>
    <div
      class="errorContainer"
    >
      <header
        class="header"
      >
        Error
      </header>
      <div
        class="content"
      >
        <p>
          Something went wrong!
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
